From 9ec6cf47a30fbdc4c64b7b0a0b4fb446f9d2904c Mon Sep 17 00:00:00 2001
From: Nate Graham <nate@kde.org>
Date: Fri, 28 Oct 2022 08:01:02 -0600
Subject: [PATCH] startplasma-wayland: Don't set GDK_SCALE and GDK_DPI_SCALE

We did this to work around a bug whereby envars would be inappropriately
propagated when going from X11 to Wayland, causing GTK apps to be scaled
to too large a size

However this no longer appears to be an issue with any of the GTK apps I
tested (Blanket, Bustle, Cheese, Chromium, Firefox, Gedit, GIMP, GNOME
Sudoku, Inkscape, Peek, Video Downloader), and keeping it causes a bug
whereby non-GTK XWayland apps that perhaps inappropriately look at
these envars to determine their scale are scaled too small.

This hack has outlived its usefulness; let's kill it.

BUG: 443215
FIXED-IN: 5.26.3


(cherry picked from commit 63de3d6f1716a6924fa5924b5a57af4abaa4460f)
---
 startkde/startplasma-wayland.cpp | 3 ---
 1 file changed, 3 deletions(-)

diff --git a/startkde/startplasma-wayland.cpp b/startkde/startplasma-wayland.cpp
index 8046156700..04875c3585 100644
--- a/startkde/startplasma-wayland.cpp
+++ b/startkde/startplasma-wayland.cpp
@@ -67,9 +67,6 @@ int main(int argc, char **argv)
     setupPlasmaEnvironment();
     runStartupConfig();
     qputenv("PLASMA_USE_QT_SCALING", "1");
-    qputenv("GDK_SCALE", "1");
-    qputenv("GDK_DPI_SCALE", "1");
-
     qputenv("XDG_SESSION_TYPE", "wayland");
 
     auto oldSystemdEnvironment = getSystemdEnvironment();
-- 
GitLab

